### PR TITLE
For a required option, when even option flag is not given throw error.

### PR DESCRIPTION
**Pull request by snow01**
For a required option, when even option flag is not given throw error. 
Fixes tests accordingly.

### CODE REVIEW

**Comment by ProTip** [comment]
I need required opts as well, can we get this merged?

On a side note, would also be great to call callbacks even if the option is missing...

**Comment by snow01** [comment]
@ProTip - Did not understand your point on allowing callbacks.

**Comment by alleles** [comment]
:scream_cat: need this!

**Comment by jjshammas** [comment]
Any update on this? Been a long time.

**Comment by snow01** [comment]
No wonder, I can see 36 pull requests pending on this wonderful project. After waiting long for this pull request to pass, I had created another module that provides similar functionality as commander and adds few more niceties. I would share soon.

**Comment by anton-rudeshko** [comment]
@zhiyelee @SomeKittens come on guys, could you please at least have a look at this PR?

**Comment by szimmer1** [comment]
bump

**Comment by JoshuaKGoldberg** [comment]
Bump

File: test/test.arguments.js [hunk 1]
```diff
@@ -15,8 +15,8 @@ program
     cmdValue = cmd;
     envValue = env;
   })
-  .option('-C, --chdir <path>', 'change the working directory')
-  .option('-c, --config <path>', 'set config path. defaults to ./deploy.conf')
+  .option('-C, --chdir [path]', 'change the working directory')
+  .option('-c, --config [path]', 'set config path. defaults to ./deploy.conf')
```
Comment (zhiyelee) [inline test/test.arguments.js]: Previously, `<>` means argument is necessary for current option which not means this option is necessary.
Correct me if I am wrong

**Comment by steve-a-jones** [comment]
Looks like this merge was rolled back .. https://github.com/tj/commander.js/pull/573 . Was this due to build failure or failed tests?

Works fine for me, but currently on

"commander": "git+https://github.com/tj/commander.js.git#975e5a42f27887c8ed23002dac6f091fbb48fe81"

**Comment by shadowspawn** [comment]
This Pull Request makes all options with required values into mandatory options that must be specified. I consider those two independent things.

See https://github.com/tj/commander.js/issues/230#issuecomment-478181355 for a long description.

**Pull request merged**